[13.0] shopinvader: Migration to 13.0

lmignon opened the pull request:
Nice to have:

* [ ] create methods should be rewritten to support multi 
* [ ] The logic to bind / unbind products and categories should be implemented as component in place of wizard. Previously it was possible to work with in-memory record of the wizard to call the same logic from within odoo. In Odoo 13 it's no more the case. That means that to rebind thousand of records we must create thousand of rows into the database to reuse the logic provided by the wizard.
* [ ] On `product.category` the name is no more translatable in V13. This functionality has been restored into shopinvader. This should be moved into a dedicated addon

lmignon committed:
[ADD] setup.py

lmignon committed:
[MIG] shopinvader: Migration to 13.0

rousseldenis reviewed (approved):
Code review

rousseldenis committed:
[13.0][FIX] shopinvader: Fix product binding

rousseldenis committed:
[13.0][FIX] shopinvader: Fix category binding

rousseldenis commented:
It seems that tests are failing due to https://github.com/odoo/odoo/pull/41791

Don't know why yet.

rousseldenis committed:
[13.0][FIX] shopinvader: Fix multi company tests

Since https://github.com/odoo/odoo/commit/af50c694445d49f1098de4236b6e8bdd9b1c93f2
it's dangerous to write company_id on user as it will be kept in cache.

To simulate company change in tests, better to add 'allowed_company_ids'
in context

rousseldenis committed:
[13.0] shopinvader: Add ROADMAP features

rousseldenis commented:
I've put missing technical features in ROADMAP

rousseldenis commented:
/ocabot merge
